components/server/src/workspace/workspace-starter.ts

```diff
@@ -510,6 +511,15 @@ export class WorkspaceStarter {
                 log.error({ userId: user.id }, "Failed to block user.", error, { contextURL });
             }
         }
+        if (blockedRepository.blockFreeUsage) {
```

Comment:
This is the actual change of the PR. 
The rest is primarily CRUD and plumbing for `blockFreeUsage`.